### PR TITLE
1st draft for table10, 11, 12

### LINKED ISSUE #101: Tables 10, 11, and 12
The 3 table functions will be very similar so should be easy to create in one PR.

## Table Layout & Design

Table 10:
table_10
Table 11:
table_11
Table 12:
table_12

Please refer to pages 17-18 of the pdf here for more details: [FDA-2022-N-1961-0002_attachment_1.pdf](https://github.com/pharmaverse/falcon/files/10744259/FDA-2022-N-1961-0002_attachment_1.pdf)

## Description

Table 10: Patients With Serious Adverse Events by System Organ Class and FDA Medical Query (Narrow), Safety Population, Pooled Analyses

Table 11: Patients with FDA Medical Query (Narrow) Leading to Treatment Discontinuation, Safety Population, Pooled Analyses

Table 12: Patients With Adverse Events Leading to Treatment Discontinuation by System Organ Class and Preferred Term, Safety Population, Pooled Analyses

## Analysis Datasets

**ADAE** - df
**ADSL** - alt_counts_df

When using data from the `scda` package, please use datasets with `archive_name` set to `"rcd_2022_10_13"`.

## Analysis Filtering

`SAFFL == "Y"` - Safety Population

## Column Variable

**ARM** (ignore "Risk Difference" column for now. This will be added in at a later date once functionality is available.)

### PR DESCRIPTION
**Pull request by YoshitoKoujin**
fda-table_10
fda-table_11
fda-table_12

Closes #101

### CODE REVIEW

**Comment by edelarua** [comment]
Hi @YoshitoKoujin,

It looks great so far! I think all the logic is correct - the only thing I changed is the criteria for "AE leading to discontinuation", from `AEACN == "DRUG WITHDRAWN"` to `DCSREAS == "ADVERSE EVENT"`, but I'm not 100% sure this is correct so if you disagree with that you can change it. I also modified the data in the examples to improve them a bit.

Could you please add tests and create templates to add to the site for each table?

Thanks!

File: quarto/table-templates/template-table_10.qmd [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+```{r tbl, message=FALSE, warning=FALSE}
+# Load Libraries & Data
+library(dplyr)
+library(falcon)
+
+adsl <- scda::synthetic_cdisc_dataset("rcd_2022_10_13", "adsl")
+adae <- scda::synthetic_cdisc_dataset("rcd_2022_10_13", "adae")
+
+# Pre-Processing - Ensure required variables FMQ01SC and FMQ01NAM exist in adae
```
Comment (edelarua) [inline quarto/table-templates/template-table_10.qmd:22]:
```suggestion
# Pre-Processing - Ensure required variables fmqsc_var and fmqnam_var exist in adae
```

File: quarto/table-templates/template-table_10.qmd [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+| `arm_var`        | (`character`) Arm variable used to split table into columns.                                                                                                                                                                                           | `"ARM"`       |
+| `fmqsc_var`      | (`character`) FMQ scope variable to use in table.                                                                                                                                                                                                      | `"FMQ01SC"`   |
+| `fmqnam_var`     | (`character`) FMQ reference name variable to use in table.                                                                                                                                                                                             | `"FMQ01NAM"`  |
+| `fmq_scope`      | (`character`) FMQ scope, must be "NARROW" or "BROAD".                                                                                                                                                                                                  | `"NARROW"`    |
```
Comment (edelarua) [inline quarto/table-templates/template-table_10.qmd:60]:
```suggestion
| `fmq_scope`      | (`character`) FMQ scope, can be `"NARROW"` or `"BROAD"`.                                                                                                                                                                                                  | `"NARROW"`    |
```

File: quarto/table-templates/template-table_10.qmd [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+| `lbl_overall`    | (`character`) If specified, an overall column will be added to the table with the given value as the column label.                                                                                                                                     | `NULL`        |
+| `prune_0`        | (`flag`) Whether all-zero rows should be removed from the table.                                                                                                                                                                                       | `FALSE`       |
+| `na_level`       | (`character`) string to represent missing values.                                                                                                                                                                                                      | `"<Missing>"` |
```
Comment (edelarua) [inline quarto/table-templates/template-table_10.qmd:63]:
```suggestion
| `na_level`       | (`character`) String to represent missing values.                                                                                                                                                                                                      | `"<Missing>"` |
```

File: quarto/table-templates/template-table_11.qmd [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+```{r tbl, message=FALSE, warning=FALSE}
+# Load Libraries & Data
+library(dplyr)
+library(falcon)
+
+adsl <- scda::synthetic_cdisc_dataset("rcd_2022_10_13", "adsl")
+adae <- scda::synthetic_cdisc_dataset("rcd_2022_10_13", "adae")
+
+# Pre-Processing - Ensure required variables FMQ01SC and FMQ01NAM exist in adae
```
Comment (edelarua) [inline quarto/table-templates/template-table_11.qmd:22]:
```suggestion
# Pre-Processing - Ensure required variables fmqsc_var and fmqnam_var exist in adae
```

File: quarto/table-templates/template-table_11.qmd [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+| `arm_var`        | (`character`) Arm variable used to split table into columns.                                                                                                                                                                                           | `"ARM"`       |
+| `fmqsc_var`      | (`character`) FMQ scope variable to use in table.                                                                                                                                                                                                      | `"FMQ01SC"`   |
+| `fmqnam_var`     | (`character`) FMQ reference name variable to use in table.                                                                                                                                                                                             | `"FMQ01NAM"`  |
+| `fmq_scope`      | (`character`) FMQ scope, must be "NARROW" or "BROAD".                                                                                                                                                                                                  | `"NARROW"`    |
```
Comment (edelarua) [inline quarto/table-templates/template-table_11.qmd:61]:
```suggestion
| `fmq_scope`      | (`character`) FMQ scope, can be `"NARROW"` or `"BROAD"`.                                                                                                                                                                                               | `"NARROW"`    |
```

File: quarto/table-templates/template-table_11.qmd [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+| `lbl_overall`    | (`character`) If specified, an overall column will be added to the table with the given value as the column label.                                                                                                                                     | `NULL`        |
+| `prune_0`        | (`flag`) Whether all-zero rows should be removed from the table.                                                                                                                                                                                       | `FALSE`       |
+| `na_level`       | (`character`) string to represent missing values.                                                                                                                                                                                                      | `"<Missing>"` |
```
Comment (edelarua) [inline quarto/table-templates/template-table_11.qmd:64]:
```suggestion
| `na_level`       | (`character`) String to represent missing values.                                                                                                                                                                                                      | `"<Missing>"` |
```

File: quarto/table-templates/template-table_12.qmd [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+::: panel-tabset
+## Spec. Screenshot
+
+![](../assets/images/screenshots/table_12.png){fig-align="center"}
+
+## Table
+
+```{r tbl, message=FALSE, warning=FALSE}
+# Load Libraries & Data
+library(dplyr)
```
Comment (edelarua) [inline quarto/table-templates/template-table_12.qmd:16]: Not used in this example so we can exclude it

```suggestion
```

File: quarto/table-templates/template-table_12.qmd [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+## Function Details
+
+### `make_table_12()`
+
+------------------------------------------------------------------------
+
+Required variables:
+
+-   **`adae`**: `SAFFL`, `USUBJID`, `AESOC`, `DCSREAS` and the variables specified by `pref_var` and `arm_var`.
```
Comment (edelarua) [inline quarto/table-templates/template-table_12.qmd:42]:
```suggestion
-   **`adae`**: `SAFFL`, `USUBJID`, `AEBODSYS`, `DCSREAS` and the variables specified by `pref_var` and `arm_var`.
```

**Review by edelarua — CHANGES_REQUESTED** [review]
Hi @YoshitoKoujin, the templates look good, just a few formatting suggestions from me below. Thanks!

**Comment by YoshitoKoujin** [comment]
> Hi @YoshitoKoujin, the templates look good, just a few formatting suggestions from me below. Thanks!

Hi @edelarua, Thank you for your review! I have updated the files. If no more update is required please let me know. I would like to merge to main. Thank you!

**Review by edelarua — APPROVED** [review]
Good to go - thanks @YoshitoKoujin!

**Pull request merged**